src/Core/Log.re

```diff
@@ -202,10 +202,26 @@ let consoleReporter = {
   };
 };
 
+// Something is broken with the [consoleReporter] on Windows -
+// it can cause the terminal to hang. This happens on CI, but
+// also reproduces intermittently when run locally.
+// Switching to a simpler formatter solves the issue - but it'd be nice to have our
+// colors back on Windows!
```

Comment:
Interesting. This is the code that emits the colors codes, so the problem is likely here: 

https://github.com/dbuenzli/fmt/blob/master/src/fmt.ml#L729-L746

Like most things OCaml it has probably not been tested much on Windows, but if we can reproduce the problem on Windows with this it shouldn't be too hard to fix. It's not much code at all, so the possible error space seems small.

Reply:
Actually, I started testing on Windows - I ran `esy run -f --checkhealth` 10x, and it still hung on the last time. So I'm not 100% sure the log colors are the culprit!

Looking at process explorer - it seems like there are still some threads running even after calling `exit`. So I'm trying to figure out how to figure out how to narrow that down. I believe they are the threads associated with the syntax client. 

I might create a `ThreadHelper` that logs out when a thread starts, the id, and when it stops, and ensure that all created threads are closed prior to exiting in the health check. might give some clues...